src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -270,8 +271,185 @@ into grouped
+                        Manage = false
+                    })
+                    .ToList();
+                dbContext.CollectionGroups.AddRange(newCollectionGroups);
+
+                // Update Group to clear AccessAll flag
+                group.AccessAll = false;
+            }
+
+            // Save changes for Step 1
+            await dbContext.SaveChangesAsync();
+
+            // Step 2: AccessAll migration for OrganizationUsers
+            var orgUsersWithAccessAll = dbContext.OrganizationUsers
+                .Where(ou => ou.OrganizationId == organizationId && ou.AccessAll == true)
+                .ToList();
+            var orgUserIdsWithAccessAll = orgUsersWithAccessAll
+                .Select(ou => ou.Id)
+                .ToList();
+
+            // Update existing CollectionUser rows
+            dbContext.CollectionUsers
+                .Where(cu =>
+                    orgUserIdsWithAccessAll.Contains(cu.OrganizationUserId) &&
+                    cu.Collection.OrganizationId == organizationId)
+                .ToList()
+                .ForEach(cu =>
+                {
+                    cu.ReadOnly = false;
+                    cu.HidePasswords = false;
+                    cu.Manage = false;
+                });
+
+            // Insert new rows into CollectionUser
+            foreach (var organizationUser in orgUsersWithAccessAll)
+            {
+                var newCollectionUsers = dbContext.Collections
+                    .Where(c =>
+                        c.OrganizationId == organizationId &&
+                        c.CollectionUsers.All(cu => cu.OrganizationUserId != organizationUser.Id))
+                    .Select(c => new CollectionUser
+                    {
+                        CollectionId = c.Id,
+                        OrganizationUserId = organizationUser.Id,
+                        ReadOnly = false,
+                        HidePasswords = false,
+                        Manage = false
+                    })
+                    .ToList();
+                dbContext.CollectionUsers.AddRange(newCollectionUsers);
+
+                // Update OrganizationUser to clear AccessAll flag
+                organizationUser.AccessAll = false;
+            }
+
+            // Save changes for Step 2
+            await dbContext.SaveChangesAsync();
+
+            // Step 3: For all OrganizationUsers with Manager role or 'EditAssignedCollections' permission
+            // update their existing CollectionUser rows and insert new rows with [Manage] = 1
+            // and finally update all OrganizationUsers with Manager role to User role
+            var managerOrgUsers = dbContext.OrganizationUsers
+                .Where(ou =>
+                    ou.OrganizationId == organizationId &&
+                    (ou.Type == OrganizationUserType.Manager ||
+                     (ou.Permissions != null && EF.Functions.Like(ou.Permissions, "%\"editAssignedCollections\":true%"))))
```

Comment:
I'm not sure if `ou.Permissions` is cleaned up if they change type, to be safe you should also check for `ou.Type == OrganizationUserType.Custom`. That's also what we do in the mssql sproc.

Reply:
They're not cleaned up, good point!